gamemodes/terrortown/gamemode/shared/sh_main.lua

```diff
@@ -11,6 +11,27 @@ local sneakSpeedSquared = math.pow(150, 2)
 
 TTT2ShopFallbackInitialized = false
 
+local callbackIdentifier = "TTT2RegisteredSWEPCallback"
+
+---
+-- Add callback for equipment and insert changes in the given equipmentTable
+-- @param string name the database-name of the equipment
+-- @param table equipmentTable the table to insert changes to
+-- @realm shared
+local function AddCallbacks(name, equipmentTable)
+	-- Make sure that on hot reloads old callbacks are removed before adding the new one
+	database.RemoveChangeCallback(ShopEditor.accessName, name, nil, callbackIdentifier)
+	database.AddChangeCallback(ShopEditor.accessName, name, nil, function(accessName, itemName, key, oldValue, newValue)
+		if not istable(equipmentTable) then
+			database.RemoveChangeCallback(ShopEditor.accessName, name, nil, callbackIdentifier)
```

Comment:
It has exactly the same problem. Im gonna open a separate PR for that. Just checked and the database module iterates the same way, that would cause an error if the underlying table gets changed.....
Ill see what I can do, but I guess the best way is to check inside the iteration if elements are still valid.

Reply:
What about just implementing the same Timer based fix, to delay this to the next frame? 
I think that would be an acceptable way to fix this here, in case we never get back to improving upon this.

Reply:
Yeah, Im gonna see tomorrow if it can be an easy overall fix and if not we can proceed with the timer fix.

Reply:
Its actually safe https://github.com/TTT-2/TTT2/blob/master/lua/ttt2/libraries/database.lua#L180-L227

I did a little test and the code creates temporary caches, that dont get changed. So even if a function removes itself, all other functions in the list are called normally.